miner/worker.go

```diff
@@ -1533,8 +1546,13 @@ func (w *worker) generateWork(params *generateParams) (*types.Block, *big.Int, e
 
 	orderCloseTime := time.Now()
 
-	blockBundles, allBundles, usedSbundles, err := w.fillTransactionsSelectAlgo(nil, work)
+	blockBundles, allBundles, usedSbundles, mempoolTxHashes, err := w.fillTransactionsSelectAlgo(nil, work)
+
+	if err != nil {
+		return nil, nil, err
+	}
 
+	err = VerifyBundlesAtomicity(work, blockBundles, allBundles, usedSbundles, mempoolTxHashes)
```

Comment:
Should we consider putting this behind a flag maybe? I think we always want to run with this check, but who knows

Reply:
int non parallel

and in your case its still 1 ms

Reply:
btw you don't need to add 1000 committed bundles.  even 100 is way to big for committed stuff.

but for used bundles its more correct